### ISSUE TITLE
Normalize operations of the Get functions

### ISSUE THREAD

**Issue by RedX2501** [jakubgarfield/Bonobo-Git-Server#515]
The Get{repo, user, team} functions will throw when using Guid and will return null when used with string. This should be normalized to returning null.

This makes writing code that checks if the repo/user/team exists easier to write since the typical usecase is to check if the input from the user is good.

**Comment by willdean** [comment]
In what circumstances would you have an invalid Guid for a repo/user/team?

**Comment by RedX2501** [comment]
It's just the use case:

You are dealing with front-end code, so you do
```
if(Repo.GetRepo("some-repo-name-from-a-form") == null) // this works fine
```
Now you are on the back end and you got a guid from somewhere
```
if(Repo.GetRepo(some_foreign_guid) == null // this will not work but will also not complain
```
Until you exercise that code path there is no indication that the code you wrote is wrong.

And I think that to catch such silly mistakes unifying both would be a good idea.

**Comment by willdean** [comment]
Hmm.  I see your point, although I'm not really sure where the some_foreign_guid might come from - why would we be writing validations of guids anyway? 

I would like code like this:  `Repo.GetRepo(guid).Name` to be correct, with an exception being thrown in the unlikely event that the guid is unknown to the system.   If  GetRepo(guid) can return null, every time you use it you should be adding a  (!= null) test.

Perhaps the real problem is that we're conflating two different operations - validating a name or a guid, and fetching a record from the database.

I do take your point about the inconsistency though - perhaps if we were to stick to the different behavious, then they should be given different names (like `First()` vs `FirstOrDefault()` in Linq)

**Comment by RedX2501** [comment]
Yeah, mayeb we should do something like `bool TryGet("name", Model repo)`. I think this is very C# and would work for the string and make clear that it is different from just `Get`.

**Comment by willdean** [comment]
:+1: I know some people don't like `out` parameters to functions, but I like the idea of following the .NET `TryParse` idiom, which is a really good approximation of what we're doing.

In practice, I don't think we'll really need the `Get(string)` method, because we want to validate those things from strings as early as possible in a very small number of places and then pass them around as Guids everywhere internally.   But it would be an option.  

At least being clear about what the contract of those various functions is will put us in a 10x better place than we were a month ago...
